frontend/src/Teacher/StudentTransfer.vue

```diff
@@ -0,0 +1,258 @@
+}
+</script>
+
+<template>
+  <div v-if="data">
+    <select v-model="semester">
+      <option v-for="item in Object.keys(data.semesters)" :key="item" :value="item">
+        {{ item }}
+      </option>
+    </select>
+
+    <select v-if="semester" v-model="subject" @change="onSubjectChanged">
+      <option v-for="item in Object.keys(data.semesters[semester])" :key="item" :value="item">
+        {{ item }}
+      </option>
+    </select>
+
+    <select v-if="subject" v-model="teacher">
+      <option
+        v-for="item in Object.keys(data.semesters[semester][subject])"
+        :key="item"
+        :value="item"
+      >
+        {{ item }}
+      </option>
+    </select>
+
+    <select v-if="teacher && classes" v-model="clazz">
+      <option
+        v-for="item in classes.filter((c) => c.teacher_username === teacher)"
+        :key="item.code"
+        :value="item.code"
+      >
+        {{ item.timeslot }} {{ item.code }}
+      </option>
+    </select>
+
+    <select v-if="clazz" v-model="student">
+      <option
+        v-for="item in classes.find((c) => c.code === clazz).students"
+        :key="item.username"
+        :value="item.username"
+      >
+        {{ item.username }} {{ item.first_name }} {{ item.last_name }}
+      </option>
+    </select>
+
+    <select v-if="student" v-model="target_class" @change="target_changed">
+      <option v-for="item in classes" :key="item.code" :value="item.code">
+        {{ item.teacher_username }} {{ item.timeslot }} {{ item.code }}
+      </option>
+    </select>
+
+    <table class="table table-hover table-stripped table-sm">
+      <tbody>
+        <tr v-for="assignment in assignments" :key="assignment.src_assignment_id">
+          <td>{{ assignment.name }}</td>
+          <td>{{ assignment.assigned_points }}</td>
+          <td>
+            <select v-if="target_class" v-model="assignment.dst_assignment_id">
+              <option
+                v-for="target in classes.find((c) => c.code === target_class).assignments"
+                :key="target.assignment_id"
+                :value="target.assignment_id"
+              >
+                {{ target.name }}
+              </option>
+            </select>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    <table v-if="result" class="table table-sm table-hover table-striped">
+      <tbody>
+        <tr v-for="item in result" :key="item.submit_id">
+          <td>{{ item.submit_id }}</td>
+          <td>{{ item.before }}</td>
+          <td>{{ item.after }}</td>
+        </tr>
+      </tbody>
+    </table>
+
+    <button
+      class="btn"
+      :class="canTransfer ? 'btn-success' : 'btn-danger'"
+      :disabled="!canTransfer"
+      @click="transfer"
+    >
+      {{ busy ? 'Transferring...' : 'Transfer' }}
+      <span v-if="!canTransfer"
+        >(
+        <span v-if="assignments.length">
+          <span v-if="notAssignedCount > 0">{{ notAssignedCount }} tasks need to be assigned</span>
```

Comment:
![image](https://github.com/user-attachments/assets/1f78bb48-a1d1-4236-8ebf-51d30fb5b12f)

It might be better to put this warning somewhere on the page, as now it only appears once you click on "Transfer" :)

Reply:
You're right. Updated in the latest forced push.